### PR TITLE
STL-1822 | Added dummy routes for post/get jobs in MockService

### PR DESCRIPTION
**Pull request by VictorDelCampo**
# Short Description
- Dummy routes created for posting and getting a job

# Changes
- Added dummy routes in mock erica
- Route for posting a job has optional parameters and successful return value is adapted for posting an est, in accordance with Julian
- Added commented values for the dev config profile, to be uncommented when testing the dummy routes locally (including flag for returning fail responses)

# Feedback
- No tests needed?
- Dummy routes do not return MockResponse, since they will be probable called from the client? MockResponse does not inherit from Response.

# How to review this Pull Request
[Link to Confluence](https://digitalservice4germany.atlassian.net/wiki/spaces/STL/pages/117637157/Development+Process#Approve-or-comment) (internal)

### CODE REVIEW

File: webapp/app/config.py
Comment (SannyNguyenHung) [inline webapp/app/config.py:98]: @JulianRoesner 

Should we maybe create a new config for this? For example:

```
class EricaVersionUpgradeConfig(DevelopmentConfig):
    USE_MOCK_API = True
    MOCK_MIN_REQUEST_COUNT = 5
    MOCK_DELIVER_FAIL_ON_POST = False
    MOCK_DELIVER_FAIL_ON_GET = False
```

Are these configurations temporary or do we want to use this in the future? Maybe it also makes sense to configure the current and next versions so we don't use "01" & "02" in the code anymore?

```
class EricaVersionUpgradeConfig(DevelopmentConfig):
    USE_MOCK_API = True
    MOCK_MIN_REQUEST_COUNT = 5
    MOCK_DELIVER_FAIL_ON_POST = False
    MOCK_DELIVER_FAIL_ON_GET = False
    CURRENT_VERSION="01"
    NEXT_VERSION="02"
```

This would also mean that we no longer explicitly specify the version in the ERICA_BASE_URL='http://0.0.0.0:8000/' but store the version in a separate configuration ERICA_VERSION="01".

Wdyt?

**Review by SannyNguyenHung — APPROVED** [review]
LGTM! Just one question :)

File: webapp/app/config.py
Comment (JulianRoesner) [inline webapp/app/config.py:102]: Personally, I think those should not be configuration values that we set for the application but rather attributes of the MockErica similar to `MockErica.eric_process_not_successful_error_occurred` because we probably want to change these during runtime (e.g. set them differently for different tests) and this is also an implementation detail that we do not need to set here.

File: webapp/app/config.py
Comment (JulianRoesner) [reply webapp/app/config.py:98]: > Are these configurations temporary or do we want to use this in the future? Maybe it also makes sense to configure the current and next versions so we don't use "01" & "02" in the code anymore?

I agree with you. Victor and I also talked about this and we probably should change it in that direction. For me, this does not have to be part of this PR because this will also need some changes to the infrastructure and we need to think about how to deploy those.
- Use ERICA_URL & ERICA_VERSION
- Keep ERICA_BASE_URL the same
- Add ERICA_URL as a new secret to the environments and ERICA_VERSION to the code

Comment (JulianRoesner) [reply webapp/app/config.py:98]: > Should we maybe create a new config for this? For example:

I agree. I would not want the default dev config to use the mocked Erica. What about:

```python
class MockedDevelopmentConfig(DevelopmentConfig):
    USE_MOCK_API = Ture
```

Imo the other options should be moved to the MockErica (s. other comment)

File: webapp/tests/elster_client/mock_erica.py
Comment (JulianRoesner) [inline webapp/tests/elster_client/mock_erica.py:380]: Wouldn't that be a 404?

File: webapp/tests/elster_client/mock_erica.py
Comment (JulianRoesner) [inline webapp/tests/elster_client/mock_erica.py:373]: Just fyi because I was thinking about it: As this is dummy code, I think that this is fine. However, in other cases I would think that we should catch a missing request_id more explicit:
```python
count = MockErica.request_id_count.get(request_id, None) + 1

if count is None:
    response = {"errorCode": -1, "errorMessage": "Request ID not found"}, 500

#....
```

Otherwise we could get errors if the error count is set to zero at some point and we assume the request id is missing

**Review by JulianRoesner — COMMENTED** [review]
MockErica looks good. Just some comments about the configuration that I think we should discuss

File: webapp/tests/elster_client/mock_erica.py
Comment (VictorDelCampo) [reply webapp/tests/elster_client/mock_erica.py:373]: I just followed the tech spec. There you only defined a 422 for errors, but yes we should use 404 for missing request id. So 422 is ok here? And also the way is handled with elif count == 0 ?

File: webapp/tests/elster_client/mock_erica.py
Comment (VictorDelCampo) [reply webapp/tests/elster_client/mock_erica.py:380]: I just followed the tech spec. There you only defined a 422 for errors, but yes we should use 404 for missing request id. So 422 is ok here?

File: webapp/app/config.py
Comment (SannyNguyenHung) [reply webapp/app/config.py]: Thats fine for me, we can change the ERICA_BASE_URL in another ticket / PR

File: webapp/tests/elster_client/mock_erica.py
Comment (VictorDelCampo) [reply webapp/tests/elster_client/mock_erica.py:383]: Changed 500 error to 404 for request id not found

**Comment by VictorDelCampo** [comment]
@JulianRoesner waiting for your approval :)

**Review by JulianRoesner — APPROVED** [review]
🚀

**Pull request merged**
